meal-planner-fredericton-ui/src/dashboard/Dashboard.jsx

```diff
@@ -220,11 +251,19 @@ const Dashboard = props => {
     </Paper>
   ) : null;
   
-
-
-
+  const [showDelete, setShowDelete] = React.useState(false);
+  const [currentRow, setCurrentRow] = React.useState(null);
+  
   return (
     <>
+      {showDelete?<DeleteDialog 
+      mealPlan={currentRow} 
+      onDelete={()=> {
+        setShowDelete(false);
+        props.relay.refetch();
+        }} 
+      onCancel={()=>setShowDelete(false)} 
+      />:""}
```

Comment:
Instead of have the delete dialog by default in the render tree and controlling the visibility with its open property, I am rendering the delete dialog only when the showDelete state is set onClick of DeleteIcon link. 
In DeleteDialog I am passing the currentRow as the mealPlan prop. onDelete prop is an inline function that removes the delete dialog and refreshes the content. It is called when delete is successful. OnCancel just removes the delete dialog.